index.js

```diff
@@ -1,176 +1,172 @@
-var inherits = require('util').inherits
-var RandomAccess = require('random-access-storage')
-var fs = require('fs')
-var mkdirp = require('mkdirp-classic')
-var path = require('path')
-var constants = fs.constants || require('constants') // eslint-disable-line
-
-var READONLY = constants.O_RDONLY
-var READWRITE = constants.O_RDWR | constants.O_CREAT
-
-module.exports = RandomAccessFile
-
-function RandomAccessFile (filename, opts) {
-  if (!(this instanceof RandomAccessFile)) return new RandomAccessFile(filename, opts)
-  RandomAccess.call(this)
-
-  if (!opts) opts = {}
-  if (opts.directory) filename = path.join(opts.directory, path.resolve('/', filename).replace(/^\w+:\\/, ''))
-
-  this.directory = opts.directory || null
-  this.filename = filename
-  this.fd = 0
-
-  // makes random-access-storage open in writable mode first
-  if (opts.writable || opts.truncate) this.preferReadonly = false
-
-  this._size = opts.size || opts.length || 0
-  this._truncate = !!opts.truncate || this._size > 0
-  this._rmdir = !!opts.rmdir
-  this._lock = opts.lock || noLock
-  this._sparse = opts.sparse || noLock
-  this._alloc = opts.alloc || Buffer.allocUnsafe
-}
+const RandomAccessStorage = require('random-access-storage')
+const fs = require('fs')
+const path = require('path')
+const constants = fs.constants || require('constants') // eslint-disable-line n/no-deprecated-api
 
-inherits(RandomAccessFile, RandomAccess)
+const READONLY = constants.O_RDONLY
+const READWRITE = constants.O_RDWR | constants.O_CREAT
 
-RandomAccessFile.prototype._open = function (req) {
-  var self = this
+module.exports = class RandomAccessFile extends RandomAccessStorage {
+  constructor (filename, opts = {}) {
+    super()
 
-  mkdirp(path.dirname(this.filename), ondir)
+    if (opts.directory) filename = path.join(opts.directory, path.resolve('/', filename).replace(/^\w+:\\/, ''))
```

Comment:
what does the regex do?

Reply:
Looks like it removes the protocol portion if passing a URL. It wasn't changed from `master`: https://github.com/random-access-storage/random-access-file/blob/58d9e1aa3fe8cc3868451a299ae272d1fd983378/index.js#L18